### PR TITLE
Improve score test documentation based on Lucas's feedback (#1835)

### PR DESCRIPTION
**Pull request by mattbowen-usds**

### CODE REVIEW

File: .github/workflows/deploy_be_staging.yml [hunk 1]
```diff
@@ -61,7 +61,10 @@ jobs:
           poetry run python3 data_pipeline/application.py score-full-run
       - name: Generate Score Post
         run: |
-          poetry run python3 data_pipeline/application.py generate-score-post -s aws
+          poetry run python3 data_pipeline/application.py generate-score-post
```
Comment (lucasmbrown-usds) [inline .github/workflows/deploy_be_staging.yml:64]: What does removing `-s aws` here do?

File: data/data-pipeline/data_pipeline/tests/score/test_output.py [hunk 2]
```diff
@@ -254,6 +254,15 @@ def test_data_sources(
         key: value for key, value in locals().items() if key != "final_score_df"
     }
 
+    # For each data source that's injected via the fixtures, do the following:
+    # * Ensure at least one column from the source shows up in the score
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/tests/score/test_output.py:258]: :tada:

File: data/data-pipeline/data_pipeline/tests/score/test_output.py [hunk 3]
```diff
@@ -301,6 +311,8 @@ def test_data_sources(
                 assert df[final_column].equals(
                     df[data_source_column]
                 ), error_message
+            # For numeric sources, use np.close so we don't get harmed by
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/tests/score/test_output.py:314]: nice comment!

File: data/data-pipeline/data_pipeline/tests/score/test_tiles_smoketests.py [hunk 4]
```diff
@@ -102,6 +110,19 @@ def test_tract_equality(tiles_df, final_score_df):
     assert tiles_df.shape[0] == final_score_df.shape[0]
 
 
+def is_col_fake_bool(col) -> bool:
```
Comment (lucasmbrown-usds) [inline data/data-pipeline/data_pipeline/tests/score/test_tiles_smoketests.py:113]: nice

**Review by lucasmbrown-usds — APPROVED** [review]
LGTM! Question about the AWS flag.

**Review by esfoobar-usds — APPROVED** [review]
🍏

File: .github/workflows/deploy_be_staging.yml (hunk 1 above)
Comment (mattbowen-usds) [reply .github/workflows/deploy_be_staging.yml:64]: uses the data that was downloaded score-full-run, which saves us a download and speeds up the pipeline a little

**Pull request merged**